### PR TITLE
Added Migration Documentation for Transform API for 0.11

### PR DESCRIPTION
**Pull request by Tattomoosa**
## Description

Added Migration Notes as discussed in #1561. They could probably use a once-over by someone who knows Amethyst, I'm new here.

## Additions
n/a

## Removals
n/a

## Modifications

Modified book with migration docs for Transform for 0.11

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Ran `cargo test --all` locally if this modified any rs files.
- [ ] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [ ] Updated the content of the book if this PR would make the book outdated.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new APIs if any were added in this PR.
- [x ] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: book/src/appendices/b_migration_notes/transform_api_changes.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+```patch
+-transform.rotate_global(Vector3::new(0.2, 0.4, 0.6), 0.4);
++transform.prepend_rotation(Vector3::new(0.2, 0.4, 0.6), 0.4);
+```
+
+### `(pitch|yaw|roll)_global` to `append_rotation_(x|y|z)_axis`
+
+```patch
+-pitch_global(0.4);
```
Comment (torkleyy) [inline book/src/appendices/b_migration_notes/transform_api_changes.md]: You’ve used `transform.` in the other places, would you mind making that consistent?

File: book/src/appendices/b_migration_notes/transform_api_changes.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+-transform.set_z(2.4);
++transform.set_translation_z(2.4);
+```
+
+```patch
+-transform.set_xyz(0.2, 1.0, 0.8);
++transform.set_translation_xyz(0.2, 1.0, 0.8);
+```
+
+### `_local` to `append_`
```
Comment (torkleyy) [inline book/src/appendices/b_migration_notes/transform_api_changes.md]: Not exactly specific to this PR, but why was this changed? The previous naming seems more obvious to me.

Comment (Tattomoosa) [reply book/src/appendices/b_migration_notes/transform_api_changes.md]: I can't comment on that, here's the relevant issue: https://github.com/amethyst/amethyst/pull/1451

File: book/src/appendices/b_migration_notes/transform_api_changes.md (hunk 1 above)
Comment (Tattomoosa) [reply book/src/appendices/b_migration_notes/transform_api_changes.md]: Not at all, that was an oversight on my part. Good catch, will fix tonight.

File: book/src/appendices/b_migration_notes/transform_api_changes.md (hunk 2 above)
Comment (torkleyy) [reply book/src/appendices/b_migration_notes/transform_api_changes.md]: I see, thanks.

File: book/src/appendices/b_migration_notes/transform_api_changes.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+-transform.yaw_global(0.4);
++transform.prepend_rotation_y_axis(0.4);
+```
+
+```patch
+-transform.roll_global(0.4);
++transform.prepend_rotation_z_axis(0.4);
+```
+
+### `translate_[xyz]` to `prepend_translation_[xyz]`
```
Comment (azriel91) [inline book/src/appendices/b_migration_notes/transform_api_changes.md]: hm, could you use either `(x|y|z)` here, or square brackets for the other points

**Comment by Tattomoosa** [comment]
I think I've resolved everything now. I went for the `###` headings initially because I felt like the separate code blocks were hard to parse otherwise, but simply combining the code blocks is definitely better. Thanks for reviewing it and improving the summary, please let me know if it needs anything else.

**Comment by azriel91** [comment]
bors r=jojolepro, torkleyy, azriel91

yeap we're good 👍
